parser.y

```diff
@@ -9037,6 +9038,32 @@ TextType:
 		$$ = x
 	}
 
+OptCharsetWithOptBinary:
+	OptBinary
+	{
+		$$ = $1
+	}
+|	"ASCII"
+	{
+		$$ = &ast.OptBinary{
+			IsBinary: false,
+			Charset:  charset.CharsetLatin1,
+		}
+	}
+|	"UNICODE"
+	{
+		$$ = &ast.OptBinary{
+			IsBinary: false,
+			Charset:  charset.CharsetUTF8,
```

Comment:
The following behaviors are specified in the [documentation](https://dev.mysql.com/doc/refman/8.0/en/string-type-overview.html) of MySQL:
> The ASCII attribute is shorthand for CHARACTER SET latin1.
The UNICODE attribute is shorthand for CHARACTER SET ucs2.

However, TiDB does not support charset `ucs2`. Should we ignore `UNICODE` in this case?
@kennytm @leoppro PTAL

Reply:
IMO it should be successfully parsed but cause TiDB itself to error with 

```
ERROR 1115 (42000): Unknown character set: 'ucs2'
```